### DISCUSSION TITLE
How to call route handler in dynamic folder - 404...

### DISCUSSION THREAD

**Discussion by clementbouakilohmydev** [vercel/next.js#58964] [App Router]
Hello, 

I have a problem calling my function in my dynamic folder

<img width="291" alt="Capture d’écran 2023-11-27 à 13 41 54" src="https://github.com/vercel/next.js/assets/127099136/ce4c4b72-50c5-4fa7-925a-a70fd873f7cb">

In `app/[lang]/contact` i have a form component where i call my function `api/contact/route.ts` but i have this

![Capture d’écran 2023-11-27 à 13 44 46](https://github.com/vercel/next.js/assets/127099136/b1a993c0-b1b2-40dc-81ea-fea92ac09873)

I tried the call this function in my `app/page.tsx` and it's worked, i think so the problem is the folder `[lang]` but even if i try to create folder `[lang]` in api folder like this `api/[lang]/contact/route.ts`, i have the same error...

Thank for your help

**Comment by icyJoseph** [comment]
Could you share the code that makes the `fetch` request? pseudo-code would be ok.

**Comment by clementbouakilohmydev** [comment]
Hello @icyJoseph , 

`app/[lang]/contact/page.tsx `
<img width="681" alt="Capture d’écran 2023-11-27 à 14 16 11" src="https://github.com/vercel/next.js/assets/127099136/47a9f676-cefa-4225-922d-ddc816a4fdef">

and this is my form component in `components/templates/contact/form.tsx`
<img width="545" alt="Capture d’écran 2023-11-27 à 14 16 50" src="https://github.com/vercel/next.js/assets/127099136/c86d7ea6-8a11-4843-a3d7-8494924d6502">

**Reply by icyJoseph** [reply]
And which one is not working? I looks like, somehow the fetch request is using `localhost:3000/fr` as the base for the fetch request, rather than just `localhost:3000` have you changed the baseURL somehow?

**Reply by clementbouakilohmydev** [reply]
`api/contact ` works on `app/page.tsx` but not `app/[lang]/contact/page.tsx` 
and `api/[lang]/contact` doesn't work at both

**Reply by clementbouakilohmydev** [reply]
i modified url with middleware to have default path with fr

**Reply by icyJoseph** [reply]
And I guess your middleware is not running for the `/` path? Could you share the matcher?

I think you'll have to exclude the `/api/contact` call from the middleware then 🤔

**Reply by clementbouakilohmydev** [reply]
`matcher: ['/((?!_next).*)']`

**Reply by NorJull** [reply]
@clementbouakilohmydev How did you solve the issue? I'm facing the same issue.
